redaxo/src/addons/backup/lib/cronjob.php

```diff
@@ -27,9 +27,47 @@ public function execute()
         if (rex_backup::exportDb($dir . $file . $ext)) {
             $message = $file . $ext . ' created';
 
+            if ($this->delete_interval) {
+                $files = glob(rex_path::addonData('backup', '*'.$ext));
+                $backups = [];
+
+                foreach ($files as $file) {
+                    $backups[$file] = filectime($file);
+                }
+                asort($backups, SORT_NUMERIC);
+
+                $limit = strtotime('-1 month'); // Generelle Vorhaltezeit: 1 Monat
+                $step = '';
+                $countDeleted = 0;
+
+                foreach ($backups as $backup => $timestamp) {
+                    if ($timestamp > $limit) {
+                        // wenn es die generelle Vorhaltezeit unterschreitet
+                        break;
+                    }
+
+                    $stepLast = $step;
+                    $step = date($this->delete_interval, (int) $timestamp);
+
+                    if ($stepLast !== $step) {
+                        // wenn es zu diesem Interval schon ein Backup gibt
+                        continue;
+                    }
+
+                    // dann löschen
+                    rex_file::delete($backup);
+                    ++$countDeleted;
+                }
+
+                if ($countDeleted) {
+                    $message .= ', '.$countDeleted.' old backups deleted';
```

Comment:
hartcodierte message

Reply:
Du meinst, wegen kein i18n? Aktuell sind die Cronjob-Log-Messages immer unübersetzt.
Beim Übersetzen wäre ja auch die Frage, in welche Sprache. Die User können ja unterschiedliche Backendsprachen haben.

Reply:
ahh stimmt. 👍